src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Legacy/LegacySyntaxNodeExtensions.NodeStack.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+
+namespace Microsoft.AspNetCore.Razor.Language.Legacy;
+
+internal static partial class LegacySyntaxNodeExtensions
+{
+    private struct NodeStack : IDisposable
+    {
+        [ThreadStatic]
+        private static SyntaxNode[]? g_nodeStack;
```

Comment:
Why not just use the object pooling types that you pulled in a while ago?

Reply:
I never pulled the object pooling types into the compiler layer.

Reply:
Looks like somebody copied the old Roslyn `ObjectPool<T>`. I can use that in the meantime, though we should probably just use the `Microsoft.Extensions.ObjectPools` package in the compiler (which is based on Roslyn's class).